lib/sqlalchemy/orm/session.py

```diff
@@ -2449,6 +2460,8 @@ def invalidate(self) -> None:
         self._close_impl(invalidate=True)
 
     def _close_impl(self, invalidate: bool) -> None:
+        if self.close_final:
```

Comment:
**mike bayer** (zzzeek) wrote:

this doc is a little troublesome because it seems to imply that "reset" behaves differently when close_resets_only is changed.   

"This method provides for same "reset-only" behavior that the :meth:`_orm.Session.close` method has provided historically, where the state of the :class:`_orm.Session` is reset as though the object were brand new, and ready to be used again.  The method may then be useful for :class:`_orm.Session` objects which set :paramref:`_orm.Session.close_resets_only` to ``False``, so that "reset only" behavior is still available from this method."

it's not that easy to explain

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4824